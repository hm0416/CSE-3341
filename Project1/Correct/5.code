program
	A(ref r, s) begin 
		r=0; 
		s=0; 
	endfunc
	ref x;
	ref y;
begin
	x=new;
	y=new;
	x=1;
	y=1;
	begin A(x, y);
	output x;
	output y;
end