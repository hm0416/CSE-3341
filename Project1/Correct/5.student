gc:1
gc:2
15
10
gc:0
