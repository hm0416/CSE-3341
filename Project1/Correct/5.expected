PROGRAM
INT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
INPUT
ID[X]
COMMA
ID[Y]
SEMICOLON
WHILE
NEGATION
LPAREN
ID[X]
ASSIGN
CONST[0]
RPAREN
BEGIN
ID[Y]
ASSIGN
ID[Y]
MULT
ID[Y]
SEMICOLON
ID[X]
ASSIGN
ID[X]
SUB
CONST[1]
SEMICOLON
ENDWHILE
SEMICOLON
OUTPUT
ID[Y]
SEMICOLON
END
